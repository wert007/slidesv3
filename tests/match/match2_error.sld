func main() {
    match 5 {
        'hello' => {
            print('expected a type error!');
        }
        true => {
            print('expected a type error!');
        }
        else => {
            print('this is the only normal thing here');
        }
    }

    match 5 {
        else => {
            print('all good!');
        }
        else => {
            print('but we cannot have multiple of these!');
        }
    }

    let b = 99 + 5;

    match b {
        b => {
            print('illegal, since b is not a constant!');
        }
        else => {
            print('hello!');
        }
    }

    match 5 {
        5 => {
            print('This is a five');
        }
        5 => {
            print('This is also a five');
        }
        5 => {
            print('This is still a five');
        }
        else => {}
    }

    match false {
        true => {
            print('We did not match all possible cases!');
        }
    }

    match print {
        else => {
            print('cannot match on system calls!');
        }
    }

    match none {
        else => {
            print('cannot match on value none!');
        }
    }
}