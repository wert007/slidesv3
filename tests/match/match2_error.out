Error in ..\tests\match\match2_error.sld:3:9: Cannot convert type string to type int.
Error in ..\tests\match\match2_error.sld:6:9: Cannot convert type bool to type int.
Error in ..\tests\match\match2_error.sld:18:17: Found multiple else cases in match statement. Only one is allowed though.
Error in ..\tests\match\match2_error.sld:26:9: Only constant expressions are allowed here.
Error in ..\tests\match\match2_error.sld:38:9: Multiple statements with the same value (5) found in match statement.
Error in ..\tests\match\match2_error.sld:41:9: Multiple statements with the same value (5) found in match statement.
Error in ..\tests\match\match2_error.sld:47:11: Not all possible values of type bool are handled. Add an `else` case maybe?
Error in ..\tests\match\match2_error.sld:53:11: Cannot match on type print. Seems like you forgot a () maybe?
Error in ..\tests\match\match2_error.sld:59:11: Cannot match on type any?. There are not enough type information about this `none` value. Maybe give it a type before?
